disco_aws_automation/disco_alarm_config.py

```diff
@@ -206,15 +226,23 @@ def _decode_section_name(section):
         Extract special alarm parameters which are stored in section name
         """
         section_segments = section.split(".")
-        if len(section_segments) == 4:
-            team, namespace, metric_name, section_hostclass = section_segments
+        # If a section has four or more segments, treat the middle segments as the metric name and
+        # the ending segment as the hostclass. This is to support some Elasticsearch metrics like
+        # 'ClusterStatus.red'
```

Comment:
Nothing good. `red` would be interpreted as the domain name and there would probably be exception somewhere along the line that would bring the whole event to a screeching halt. We can accept that risk, change the parsing logic (and all the configs) to use a different delimiter, or we could rewrite the disco_alarms_config logic to understand a whitelist of all metric names by namespace and make sure that the metric matches that list before proceeding.

The whitelist thing might not be super hard to do, but I'm not sure if that level of effort would be in-scope for this change: http://boto3.readthedocs.io/en/latest/reference/services/cloudwatch.html#CloudWatch.Client.list_metrics